### PR TITLE
JAMES-2997 Add hasAttachment to MessageFastViewPrecomputedProperties

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/projections/CassandraMessageFastViewProjectionModule.java [hunk 1]
```diff
@@ -17,25 +17,29 @@
...
-import org.apache.james.backends.cassandra.utils.CassandraConstants;
-import org.apache.james.jmap.cassandra.preview.table.CassandraMessagePreviewTable;
 
 import com.datastax.driver.core.schemabuilder.SchemaBuilder;
 
-public interface CassandraMessagePreviewModule {
-    CassandraModule MODULE = CassandraModule.table(CassandraMessagePreviewTable.TABLE_NAME)
+public interface CassandraMessageFastViewProjectionModule {
+    CassandraModule MODULE = CassandraModule.table(TABLE_NAME)
         .comment("Storing the JMAP preview property extracted from message bodies")
```
Comment (trantienduchn) [inline server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/projections/CassandraMessageFastViewProjectionModule.java]: please update this comment

File: server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/projections/table/CassandraMessageFastViewProjectionTable.java [hunk 2]
```diff
@@ -17,11 +17,12 @@
  * under the License.                                           *
  ****************************************************************/
 
-package org.apache.james.jmap.cassandra.preview.table;
+package org.apache.james.jmap.cassandra.projections.table;
 
-public interface CassandraMessagePreviewTable {
+public interface CassandraMessageFastViewProjectionTable {
     String TABLE_NAME = "message_preview";
```
Comment (trantienduchn) [inline server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/projections/table/CassandraMessageFastViewProjectionTable.java]: s/message_preview/message_... ?

File: server/data/data-jmap/src/main/java/org/apache/james/jmap/api/projections/MessageFastViewProjection.java [hunk 3]
```diff
@@ -17,16 +17,16 @@
...
  ****************************************************************/
 
-package org.apache.james.jmap.api.preview;
+package org.apache.james.jmap.api.projections;
 
 import org.apache.james.mailbox.model.MessageId;
 import org.reactivestreams.Publisher;
 
-public interface MessagePreviewStore {
+public interface MessageFastViewProjection {
```
Comment (trantienduchn) [inline server/data/data-jmap/src/main/java/org/apache/james/jmap/api/projections/MessageFastViewProjection.java]: I prefer `MessageFastViewPropertiesStore`, I don't think Projection is the right term

Comment (chibenwa) [reply server/data/data-jmap/src/main/java/org/apache/james/jmap/api/projections/MessageFastViewProjection.java]: The content of your `Store` is a projection of the actual mailbox message content.

`Projection` is thus a more specific term.

Moreover `building a projection` is a common NOSQL wording.

More common, more specific, then why being reluctant adopting it?

**Pull request merged**